### PR TITLE
update some docs about axes/grids

### LINKED ISSUE #1721: documentation: user guide describes unsupported axis locations
http://bokeh.pydata.org/en/latest/docs/user_guide/objects.html#axes

> location: where along the cross-dimension to locate this axis: "min", "max", "left", "right", "top", "bottom" or a floating point value

I am unable to set an axis location that is not one of the four values from the Location enum - anything else (including 'min', 'max', or floating point values) yields:

> ValueError: expected an element of either Auto or Enum('above', 'below', 'left', 'right')

Is the documentation wrong? If not, how would one go about offsetting an axis?

### PR DESCRIPTION
**Pull request by bryevdv**
issues: fixes #1721

Also made some minor corrections removing references to "curplot". However, UG needs a more rigorous pass over it.

### CODE REVIEW

**Comment by bryevdv** [comment]
@bokeh/core trivial change, will merge when Travis passes

File: sphinx/source/docs/user_guide/objects.rst [hunk 1]
```diff
@@ -186,17 +185,18 @@ properties, for the axis label and major tick labels. These are prefixed ``axis_
...
-    axis.bounds = (3, 7)
-    axis.major_label_orientation = pi/4
+    ax.axis_line_color = "red"
+    ax.bounds = (3, 7)
+    ax.major_label_orientation = pi/4
 
-Axes for the current plot may be conveniently obtained using the :func:`bokeh.plotting.xaxis`, :func:`bokeh.plotting.yaxis`,
-and :func:`bokeh.plotting.axis` functions. These return collections of axes that can be indexed to retrieve
-individual axes, or can that have attributes set directly on them to update all axes. Examples::
+Axes for a may be conveniently obtained using the ``xaxis``, ``yaxis``,
```
Comment (damianavila) [inline sphinx/source/docs/user_guide/objects.rst]: `Axes for a may be`...? Is this actually correct wording? Sorry if it is, but does not look common to me...

**Comment by damianavila** [comment]
Just a minor comment, otherwise looks good.

File: sphinx/source/docs/user_guide/objects.rst (hunk 1 above)
Comment (bryevdv) [inline sphinx/source/docs/user_guide/objects.rst]: no thats a typo

**Comment by damianavila** [comment]
OK, LGTM now...

**Pull request merged**